syntax = "proto3";

package transport;


message Parameters {
    repeated bytes tensors = 1;
    string tensor_type = 2;
}

message ClientRequest {
    string message = 1;
    Parameters parameters = 2;
}

message ServerReply {
    string message = 1;
    Parameters parameters = 2;
}

service Transport {
    // Both Streaming
    rpc Join (stream ClientRequest) returns (stream ServerReply);
}